node/network/availability-distribution/src/requester/fetch_task/mod.rs

```diff
@@ -203,7 +203,9 @@ impl FetchTask {
 	/// Remove leaves and cancel the task, if it was the last one and the task has still been
 	/// fetching.
 	pub fn remove_leaves(&mut self, leaves: &HashSet<Hash>) {
-		self.live_in.difference(leaves);
+		for leaf in leaves {
+			self.live_in.remove(leaf);
```

Comment:
was that a leak?

Reply:
[Yes](https://github.com/paritytech/polkadot/blob/e0f4df5a9a1d4a6da2098ae7fd9b994eeeef44f3/node/network/availability-distribution/src/requester/mod.rs#L135). :grimacing: